src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisition.cs

```diff
@@ -196,9 +196,16 @@ public async Task<AcquireTokenResult> AddAccountToCacheFromAuthorizationCodeAsyn
             }
         }
 
+
+        /// <summary>
+        /// Allows creation of confidential client applications targeting regional and global authorities
+        /// when supporting managed identities.
+        /// </summary>
+        /// <param name="mergedOptions"></param>
+        /// <returns></returns>
         private static string GetApplicationKey(MergedOptions mergedOptions)
         {
-            return mergedOptions.Instance! + mergedOptions.ClientId;
+            return DefaultTokenAcquirerFactoryImplementation.GetKey(mergedOptions.Authority, mergedOptions.ClientId, mergedOptions.AzureRegion);
```

Comment:
This looks like a functional code difference, is there missing test coverage?

Reply:
Added a test case for it.